cmd/kubernetes.go

```diff
@@ -0,0 +1,818 @@
+			files = append(files, client.FilePair{Src: filepath.Join(config.ManagerDir + "/key.pem"), Dest: podBinDir + ".wr_" + config.Deployment + "/"})
+			files = append(files, client.FilePair{Src: filepath.Join(config.ManagerDir + "/ca.pem"), Dest: podBinDir + ".wr_" + config.Deployment + "/"})
+			files = append(files, client.FilePair{Src: filepath.Join(config.ManagerDir + "/cert.pem"), Dest: podBinDir + ".wr_" + config.Deployment + "/"})
+
+			info(fmt.Sprintf("podConfigFiles: %#v", podConfigFiles))
+
+			// Specify deployment options
+			c.Opts = &kubedeployment.DeployOpts{
+				ContainerImage:  containerImage,
+				TempMountPath:   podBinDir,
+				Files:           files,
+				BinaryPath:      podScriptDir + scriptName,
+				BinaryArgs:      binaryArgs,
+				ConfigMapName:   configMapName,
+				ConfigMountPath: podScriptDir,
+				RequiredPorts:   []int{mp, wp},
+				Logger:          appLogger,
+				ResourcePath:    resourcePath,
+			}
+
+			// Create the deployment if an existing one does not exist
+			if kubeDeploy {
+
+				info("creating WR deployment")
+				err = c.Client.Deploy(c.Opts.ContainerImage, c.Opts.TempMountPath, c.Opts.BinaryPath, c.Opts.BinaryArgs, c.Opts.ConfigMapName, c.Opts.ConfigMountPath, c.Opts.RequiredPorts)
+				if err != nil {
+					die(fmt.Sprintf("failed to create deployment: %s", err))
+				}
+			}
+
+			// Start Controller
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			info("starting controller")
+			c.Run(stopCh)
+		}
+
+	},
+}
+
+// teardown sub-command deletes all kubernetes resources we created and then stops
+// the daemon by killing it's pid.
+var kubeTearDownCmd = &cobra.Command{
+	Use:   "teardown",
+	Short: "Delete all kubernetes resources that deploy created",
+	Long: `Immediately stop the remote workflow manager, saving its state.
+
+Deletes all kubernetes resources that wr created (pods, deployments, config maps, namespaces).
+(Except for any files that were saved to persistent cloud storage.)
+
+Note that any runners that are currently running will die, along with any
+commands they were running. It is more graceful to issue 'wr manager drain'
+first, and regularly rerun drain until it reports the manager is stopped, and
+only then request a teardown (you'll need to add the --force option). But this
+is only a good idea if you have configured wr to back up its database to S3, as
+otherwise your database going forward will not reflect anything you did during
+that kubernetes deployment.
+
+If you don't back up to S3, the teardown command tries to copy the remote
+database locally, which is only possible while the remote server is still up
+and accessible.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		kubeLogger := log15.New()
+		logLevel := log15.LvlWarn
+		if kubeDebug {
+			logLevel = log15.LvlDebug
+		}
+		kubeLogger.SetHandler(log15.LvlFilterHandler(logLevel, l15h.CallerInfoHandler(log15.StderrHandler)))
+		// before stopping the manager, make sure we can interact with the
```

Comment:
Can we have just 1 bit of kubeLogger setup code in this file? Maybe a new func, or in init, or a global?